helloworld/notebooks/OpIris.ipynb

```diff
@@ -0,0 +1,337 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Everything we’ve done so far has been purely at the level of definitions. We have defined how we would like to extract our raw features from data of type `Iris`, and we have defined how we would like to manipulate them. In order to actually manifest the data described by these features, we need to add them to a workflow and attach a data source to the workflow:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "implicit val spark = SparkSession.builder.config(conf).getOrCreate()\n",
+    "import spark.implicits._ // Needed for Encoders for the Passenger case class\n",
+    "import com.salesforce.op.readers.DataReaders\n",
+    "\n",
+    "val trainFilePath = \"/home/beakerx/helloworld/src/main/resources/IrisDataset/iris.data\"\n",
+    "    // Define a way to read data into our Passenger class from our CSV file\n",
+    "val trainDataReader = DataReaders.Simple.csvCase[Iris](\n",
+    "      path = Option(trainFilePath)\n",
+    "      //key = _.id.toString\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    " Workflow for TransmogrifAI. Takes the final features that the user wants to generate as inputs and constructs the full DAG needed to generate them from those features lineage. Then fits any estimators in the pipeline dag to create a sequence of transformations that are saved in a workflow model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val workflow = new OpWorkflow().setResultFeatures(prediction, labels).setReader(trainDataReader)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When we now call ‘train’ on this workflow, it automatically computes and executes the entire DAG of Stages needed to compute the features  fitting all the estimators on the training data in the process. Calling score on the fitted workflow then transforms the underlying training data to produce a DataFrame with the all the features manifested. The score method can optionally be passed an evaluator that produces metrics.\n",
+    "`workflow.train()` methods fits all of the estimators in the pipeline and return a pipeline model of only transformers. Uses data loaded as specified by the data reader to generate the initial data set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val fittedWorkflow = workflow.train()\n",
+    "println(s\"Summary: ${fittedWorkflow.summary()}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After model has been fitted we use scoreAndEvaluate() function to evaluate the metrics"
```

Comment:
it would be nice to mention that you can change out the data before doing this by either setting a new path or a new reader

Reply:
changed